superset/common/utils/query_cache_manager.py

```diff
@@ -145,7 +145,7 @@ def get(
         Initialize QueryCacheManager by query-cache key
         """
         query_cache = cls()
-        if not key or not _cache[region] or force_query:
+        if not key or not _cache[region] or not force_query:
```

Comment:
This change means that if **force_query is false**, then generate a new query_cache object.....after this change, we can't use cached data any more.

Reply:
@zhaoyongjie hi,
Thank you for having a look.
Likely I am missing something, to clarify: we will use cached data in all cases which satisfy above "if statement". The only change made that we will not use cached data when a user forces a query (both for synch and async queries). Is not it what we want? 

E.g. when a user forces a query, data should not be extracted from a cache, but from the source (e.g. a database) and a new cache object created (based on forced query to a source). I understand that it will increase load on a source but otherwise we are loosing the sense of using "force query" and this functionality unlikely should be abused so load on source should not change significantly.

Reply:
@GAlexeyV, the default value of force_query is False(the frontend side also emit **false** as force_query), and the **not force_query** condition in **or** expression means that all of default queries(except true as force_query) emit to Superset should create a new "query_cache" object --- please don't use cached data. It is an opposite behavior to we want, what is the expected behavior before? 1. if the data cached 2. and the force_query is False 3. then the query should been responded by the Cache.

You could set a stop debugger at line 149 to verify this situation.

Reply:
@zhaoyongjie, thank you for explaining. Struggling to understand the codebase. Submitted a different change: removed a check for cached data for async queries when cache is forced.

Reply:
@zhaoyongjie hi,
Any chance you can have a look?

Reply:
@john-bodley hi,
Maybe you can have a look?
Will update tests if you ok with the change

Reply:
@GAlexeyV I'll check out this part of code this week as well, but for now the CI seems broken.

Reply:
@zhaoyongjie , got it, thank you
Yes, I have not updated tests as firstly wanted to agree on change

Reply:
@GAlexeyV Sorry for the waiting, I'll review this tonight(CET timeone)

Reply:
Cool, thanks

Reply:
@GAlexeyV I put some thoughts in your codes, thanks a lot for the fix. ping me if you need supports.

Reply:
@zhaoyongjie , replied to a comment